components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/internal/EntitlementServiceComponent.java

```diff
@@ -192,6 +197,23 @@ protected void activate(ComponentContext ctxt) {
             builder.setBundleContext(ctxt.getBundleContext());
             builder.buildEntitlementConfig(EntitlementConfigHolder.getInstance());
 
+            boolean balanaConfig = Boolean.parseBoolean((String) EntitlementServiceComponent.getEntitlementConfig().
+                    getEngineProperties().get(PDPConstants.BALANA_CONFIG_ENABLE));
+            
+            String configProperty = System.getProperty(PDP_CONFIG_FILE_PATH);
+
+            if (balanaConfig && configProperty == null) {
```

Comment:
What happen if balanaConfig is false and configProperty = null ?

Reply:
@GayanM then Balana uses its default config file.